code/modules/hydroponics/grown/root.dm

```diff
@@ -28,10 +28,18 @@
 /obj/item/food/grown/carrot/attackby(obj/item/I, mob/user, params)
 	if(I.get_sharpness())
 		to_chat(user, span_notice("You sharpen the carrot into a shiv with [I]."))
-		var/obj/item/knife/shiv/carrot/Shiv = new /obj/item/knife/shiv/carrot
+		var/Cblade
+		if(seed.potency > 30)
+			var/carrot_sword_chance = (seed.potency - 30) * 5
```

Comment:
Because of the way the math adds up, you don't need to do `if potency > 30`. Because if it was less than 30, it would be a probability of 0.